cmd/kubelet/app/server.go

```diff
@@ -488,13 +493,19 @@ func (s *KubeletServer) kubeconfigClientConfig() (*client.Config, error) {
 // fall back to the default auth (none) without an error.
 // TODO(roberthbailey): Remove support for --auth-path
 func (s *KubeletServer) createClientConfig() (*client.Config, error) {
-	if s.KubeConfig.Provided() && s.AuthPath.Provided() {
-		return nil, fmt.Errorf("cannot specify both --kubeconfig and --auth-path")
-	}
-	if s.KubeConfig.Provided() {
-		return s.kubeconfigClientConfig()
-	} else if s.AuthPath.Provided() {
-		return s.authPathClientConfig(false)
+	authPathFlag := pflag.Lookup(authPathFlagName)
+	kubeconfigFlag := pflag.Lookup(kubeconfigFlagName)
+	if authPathFlag != nil && kubeconfigFlag != nil {
+		if authPathFlag.Changed && kubeconfigFlag.Changed {
```

Comment:
So a 'fix' might be to move the logic to get the client.Config correct before the call to createClientConfig? Wherein we fix fill in the struct correctly before we get here, based on both the cmd() and the saved config?  If you want to reuse 'createClientConfig' you have to pass it valid value?  If you really want the 'unset' to live in the client.Config we can put it next to the Bool/String values and set it explicitly when we are parsing the cmd....